src/main/java/hanieum/conik/domain/address/CompanyAddress.java

```diff
@@ -0,0 +1,23 @@
+package hanieum.conik.domain.address;
+
+import hanieum.conik.domain.address.dto.AddressRegisterRequest;
+import jakarta.persistence.Entity;
+import lombok.Getter;
+
+@Entity
+@Getter
+public class CompanyAddress extends AddressBase {
+    protected CompanyAddress() { super(null,null,null); }
+
+    private CompanyAddress(String postal, String street, String detail) {
+        super(postal, street, detail);
+    }
+
+    public static CompanyAddress register(AddressRegisterRequest req) {
+        return new CompanyAddress(
+                req.addressPostalCode(),
+                req.addressStreetAddress(),
+                req.addressDetailAddress()
+        );
+    }
+}
```

Comment:
```suggestion
@Entity
@Getter
@NoArgsConstructor(callSuper = true, access = AccessLevel.PROTECTED)
@AllArgsConstructor(callSuper = true, access = AccessLevel.PRIVATE)
public class CompanyAddress extends AddressBase {

    public static CompanyAddress register(AddressRegisterRequest req) {
        return new CompanyAddress(
                req.addressPostalCode(),
                req.addressStreetAddress(),
                req.addressDetailAddress()
        );
    }
}
```
lombok을 쓰고 있으고 다른 엔티티들에서는 @NoArgs나 @AllArgs를 사용해서 일관성 있게 애노테이션으로 처리할 수도 있을 것 같아요

Reply:
@SeongHo5356 @NoArgsConstructor·@AllArgsConstructor 에는 callSuper 옵션이 따로 정의되어있지 않은 것 같은데 직접 만들어서 사용하셨나요?

Reply:
[참고](https://sjparkk-dev1og.tistory.com/m/84)

롬복에서 기본적으로 제공하는 속성으로 보이는데 적용이 안되나요?!

Reply:
@SeongHo5356 저 생성자 어노테이션에는 없는 속성인 것 같아요!

Reply:
아하 그러면 이대로가 최선인 것 같네요!!👍